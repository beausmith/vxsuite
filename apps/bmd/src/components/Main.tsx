import styled from 'styled-components'

interface Props {
  noMargin?: boolean
}

const Main = styled('main')<Props>`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  flex: 1;
  text-align: center;
  margin: ${({ noMargin }: Props) => (noMargin ? undefined : '1rem')};
  @media print {
    justify-content: flex-start;
    margin: 0;
  }
`

export default Main
