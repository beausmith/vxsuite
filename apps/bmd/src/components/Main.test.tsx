import { axe } from 'jest-axe'
import React from 'react'
import { render } from 'react-testing-library'

import Main from './Main'

it(`renders Main`, async () => {
  const { container } = render(<Main>foo</Main>)
  expect(await axe(container.innerHTML)).toHaveNoViolations()
  expect(container.firstChild).toMatchSnapshot()
})
