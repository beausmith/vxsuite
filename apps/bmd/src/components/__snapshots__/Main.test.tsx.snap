// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Main 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
  margin: 1rem;
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    margin: 0;
  }
}

<main
  class="c0"
>
  foo
</main>
`;
